pinot-spi/src/main/java/org/apache/pinot/spi/env/CommonsConfigurationUtils.java

```diff
@@ -85,13 +87,18 @@ public static PropertiesConfiguration fromPath(String path)
   public static PropertiesConfiguration fromPath(@Nullable String path, boolean setDefaultDelimiter,
       @Nullable PropertyIOFactoryKind ioFactoryKind)
```

Comment:
Can we also include `boolean failOnDuplicateKeys` as a parameter so that it is clear to the caller that no duplicate keys are expected?